src/lib/comms/sol-lwm2m.c

```diff
@@ -3606,6 +3660,27 @@ should_dispatch_notifications(uint8_t code, bool is_execute)
     return false;
 }
 
+static int64_t
+get_server_id_by_link_addr(const struct sol_vector *connections,
```

Comment:
Change this function. you should return an int here. 0 if you find the server or -ENOENT if the server was not found.
than you change the signature to.
```C
int get_server_id_by_link_addr(const struct sol_vector *connections,  const struct sol_network_link_addr *cliaddr, int64_t *TheServerIdToBeFilled);
```

Then insde the function you do 
```C
*TheServerIdToBeFilled = conn_ctx->server_id;
return 0;
```

Please rename *TheServerIdToBeFilled to something else :]